src/matter/interaction/InteractionMessenger.ts

```diff
@@ -206,8 +209,42 @@ export class InteractionClientMessenger extends InteractionMessenger<MatterContr
         return this.request(MessageType.ReadRequest, TlvReadRequest, MessageType.ReportData, TlvDataReport, readRequest);
     }
 
-    sendSubscribeRequest(subscribeRequest: SubscribeRequest) {
-        return this.request(MessageType.SubscribeRequest, TlvSubscribeRequest, MessageType.SubscribeResponse, TlvSubscribeResponse, subscribeRequest);
+    async sendSubscribeRequest(subscribeRequest: SubscribeRequest) {
+        await this.exchange.send(MessageType.SubscribeRequest, TlvSubscribeRequest.encode(subscribeRequest));
+        const initialSubscriptionValues = await this.readDataReport();
+        if (!initialSubscriptionValues.values) initialSubscriptionValues.values = [];
+        const subscriptionId = initialSubscriptionValues.subscriptionId;
+        if (subscriptionId === undefined) {
+            await this.sendStatus(StatusCode.Failure);
```

Comment:
This should be done by the catch-all method

Reply:
there is no catch all for client usage right now, so stays as is